Combined DraftRegistration and DraftNode API Endpoints

UdayVarkhedkar opened the pull request:
For group code review

UdayVarkhedkar committed:
Adds model tests for the DraftNode model changes

Commit:
Merging branches together

Commit:
Merging branches together

Commit:
Fixing tests

Commit:
Fixing more tests

Commit:
Fixing argument ordering

Commit:
Fixing order of args

Commit:
Fixing tests, adding if statement for non existent title for schemas

Commit:
Bumping schema version to prevent 400 error in tests

Commit:
Fixing hairy schema version things.

Commit:
Pulling in some schema blocks things

Commit:
Pulling in py3/Schema blocks pt 1

Commit:
Pulling in files

Commit:
Test fixing

Commit:
Fixing tests, serializers

Commit:
Fixing base serializer

Commit:
More base serializer stuff

Commit:
Pulling in changes

Commit:
Pulling in py3 changes

Commit:
Pulling in draft_registration tests

Commit:
Fixing merge errors

Commit:
Fixing merge errors

Commit:
Ooops.

Commit:
Fixing py3 related error

Commit:
Adding back in registration_responses

Commit:
Adding back in some anonymizers

Commit:
Fixing some py3 related issues

Commit:
Fixing some more test things. Py3 is awesome

Commit:
Removing ApiTestCase

Commit:
Fixing some Py3 related issues in tests

Commit:
Fixing more py3 errors

UdayVarkhedkar committed:
Initial commit pulling out DraftRegistrationRelationship endpoints code from Dawn's PR

Commit:
Final (hopefully) fix

Commit:
Pulling in serializers.py

Commit:
Pulling in test fixes

Commit:
Repairing a test

Commit:
Python 3 fixes

Commit:
Adding py3 issues

UdayVarkhedkar committed:
Removes outdated migration files

Commit:
Making recommended fixes/changes

Johnetordoff committed:
try fix for monthly data script

Commit:
Fixing mistake. Removing check permissions arg

Commit:
Putting back in something that shouldn't have been removed

Johnetordoff committed:
fix csv encoding problem

Commit:
Fixing more mistakes

UdayVarkhedkar committed:
Modifies function calls in draft_registrations/views to use super()

UdayVarkhedkar committed:
Fixes leftover merge conflicts

Commit:
Making recommended fixes. Adding functionality to update draft registration fields on POST

Commit:
Making a change suggested by Uday

mfraezz committed:
Merge branch 'hotfix/20.2.5'

mfraezz committed:
Merge branch 'hotfix/20.2.5' into develop

Commit:
Ooops

UdayVarkhedkar committed:
Adds unit tests for draft registration relationship endpoints

UdayVarkhedkar committed:
Fixes syntax issues

Commit:
Fixing things that were messed up during rebase

Commit:
Removing unncessary migration

brianjgeiger committed:
Update changelog and package.json

brianjgeiger committed:
Merge tag '20.3.0' into develop

20.3.0 Release

brianjgeiger committed:
Merge branch 'release/20.3.0'

UdayVarkhedkar committed:
Renames testing files for draft registration subjects

Commit:
Fixing issues from rebase

Commit:
Fixing a few things

Commit:
Fixing some more things

Commit:
Fixing schema blocks things that were accidentally removed during rebase

Commit:
Fixing rebase errors

Commit:
Fixing more mistakes. Getting there

Commit:
Adding back in registration responses

UdayVarkhedkar commented on api/base/urls.py:
```diff
@@ -35,11 +35,12 @@
                 url(r'^collections/', include('api.collections.urls', namespace='collections')),
                 url(r'^comments/', include('api.comments.urls', namespace='comments')),
                 url(r'^docs/', RedirectView.as_view(pattern_name=views.root), name='redirect-to-root', kwargs={'version': default_version}),
+                url(r'^draft_nodes/', include('api.draft_nodes.urls', namespace='draft-nodes')),
```
Change namespace to snake_case

corbinSanders replied:
👌

UdayVarkhedkar commented on api/base/versioning.py:
```diff
@@ -10,12 +10,13 @@
 
+# Determines the API version that will start using new
+# fields to create a registration
+CREATE_REGISTRATION_FIELD_CHANGE_VERSION = '2.19'
```
Add line after

corbinSanders replied:
👌

brianjgeiger commented on api/draft_registrations/permissions.py:
```diff
@@ -0,0 +1,51 @@
+class IsContributorOrAdminContributor(permissions.BasePermission):
+    """
+    Need to be a contributor on the draft registration to view, otherwise,
+    you need to be an admin contributor to make edits
```
"you need to be an admin contributor" on what?

UdayVarkhedkar commented on api/draft_registrations/permissions.py:
```diff
@@ -0,0 +1,51 @@
+class IsContributorOrAdminContributor(permissions.BasePermission):
+    """
+    Need to be a contributor on the draft registration to view, otherwise,
+    you need to be an admin contributor to make edits
```
```
Need to be a contributor on the draft registration to view. 
Need to be an admin contributor on the draft registration to make edits.
```

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/draft_registrations/permissions.py:
```diff
@@ -0,0 +1,51 @@
+from osf.utils import permissions as osf_permissions
+
+
+class IsContributorOrAdminContributor(permissions.BasePermission):
```
Replace with `ContributorOrPublic` (api/nodes/permissions.py) or overwrite `ContributorOrPublic` as necessary.

UdayVarkhedkar commented on api/draft_nodes/permissions.py:
```diff
@@ -0,0 +1,23 @@
+from api.base.utils import assert_resource_type
+
+
+class ContributorOnDraftRegistration(IsContributorOrAdminContributor):
```
Adapt to changes on api/draft_registration/permissions.py

UdayVarkhedkar replied:
Look into possibly modifying api.nodes.permissions.ContributorOrPublic to not require a child permissions class.

brianjgeiger commented on api/base/serializers.py:
```diff
@@ -202,6 +202,19 @@ def should_be_none(self, instance):
         return not isinstance(self.field, RelationshipField)
 
 
+class HideIfDraft(ConditionalField):
```
Minor: Would be better as "HideIfDraftRegistration" unless we want to use this for other draft types. If that's the case, we should generalize or add a todo to generalize.

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/draft_nodes/serializers.py:
```diff
@@ -0,0 +1,49 @@
+from api.nodes.serializers import NodeStorageProviderSerializer
+
+
+class DraftNodeSerializer(JSONAPISerializer):
```
Have a discussion about alternative/better naming for DraftNode ('FileBasket') within the API (to serve as a layer of abstraction over the DraftNode model)

UdayVarkhedkar reviewed (commented):
Go through and review uses of `list(items())` resulting from python3 modifications

UdayVarkhedkar commented on api/base/serializers.py:
```diff
@@ -857,20 +870,17 @@ def to_representation(self, value):
         self_url = url['self']
         self_meta = self.get_meta_information(self.self_meta, value)
         relationship = format_relationship_links(related_url, self_url, related_meta, self_meta)
```
Investigate changes and ensure that necessary logic wasn't removed during rebasing of branch with schemablocks changes

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/base/serializers.py:
```diff
@@ -983,18 +997,7 @@ def to_representation(self, value):
         """
         meta = functional.rapply(self.meta, _url_val, obj=value, serializer=self.parent, request=self.context['request'])
         obj = getattr(value, 'referent', value)
-        return {
```
Add back in - removed as a result of the rebase after schemablocks was merged into develop

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/draft_registrations/views.py:
```diff
@@ -1,49 +1,220 @@
-    required_write_scopes = [CoreScopes.NODE_DRAFT_REGISTRATIONS_WRITE]
+    serializer_class = DraftRegistrationSerializer
+
+    view_category = 'draft_registrations'
```
Change all view_categories to kebab-case

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/draft_registrations/views.py:
```diff
@@ -1,49 +1,220 @@
 
     serializer_class = DraftRegistrationDetailSerializer
+
     view_category = 'draft_registrations'
```
change to kebab case

UdayVarkhedkar commented on api/draft_registrations/views.py:
```diff
@@ -1,49 +1,220 @@
+    view_category = 'draft_registrations'
+    view_name = 'draft-registration-relationships-institutions'
 
-            if draft.registered_node and not draft.registered_node.is_deleted:
```
Ensure tests for the removed Exceptions raises

UdayVarkhedkar replied:
Moved to DraftMixin

brianjgeiger commented on api/nodes/serializers.py:
```diff
@@ -1194,9 +1189,11 @@ def get_absolute_url(self, obj):
 
     def get_unregistered_contributor(self, obj):
-        # SerializerMethodField works for both Node and Preprint contributors
+        # SerializerMethodField works for both Node/DraftRegistration/Preprint contributors
```
Super-minor: technically not 'both' anymore.

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/draft_registrations/serializers.py:
```diff
@@ -0,0 +1,215 @@
+
+class DraftRegistrationDetailSerializer(DraftRegistrationSerializer, DraftRegistrationDetailLegacySerializer):
+    """
+    Overrides DraftRegistrationSerializerLegacy to make id required.
```
Make doc string more descriptive

UdayVarkhedkar replied:
Add tests to ensure that the id is required

corbinSanders replied:
The ID would be given in the query parameter in the URL. If theres no ID given, it would just be the draft registration list endpoint.

UdayVarkhedkar commented on api/nodes/serializers.py:
```diff
@@ -1377,11 +1374,6 @@ class NodeStorageProviderSerializer(JSONAPISerializer):
         'new_folder': WaterbutlerLink(kind='folder'),
         'storage_addons': 'get_storage_addons_url',
     })
-    root_folder = RelationshipField(
```
Add back - removed during schemablocks merge

corbinSanders replied:
🆗

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/nodes/serializers.py:
```diff
@@ -1533,12 +1525,14 @@ def enforce_metadata_or_registration_responses(self, metadata=None, registration
+        node = self.get_node(validated_data)
         # Old workflow - deeply nested
         metadata = validated_data.pop('registration_metadata', None)
-        # New workflow - less nesting
```
Add back - removed in merge with schemablocks

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/nodes/serializers.py:
```diff
@@ -1549,12 +1543,20 @@ def create(self, validated_data):
         if metadata:
-            self.update_metadata(draft, metadata, reviewer)
-
+                try:
```
Fix indentation

UdayVarkhedkar replied:
Replace with `self.update_metadata(draft, metadata, reviewer)`

corbinSanders replied:
🆗

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/nodes/serializers.py:
```diff
@@ -1566,15 +1568,12 @@ def get_type(request):
 
     Also makes registration_supplement read-only.
-
-    Either pass in registration_metadata (old workflow) or registration_responses
```
Add back - removed in merge w/ schemablocks

corbinSanders replied:
🆗

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/nodes/serializers.py:
```diff
@@ -1600,7 +1599,13 @@ def update(self, draft, validated_data):
 
         if metadata:
-            self.update_metadata(draft, metadata, reviewer)
+            try:
```
Replace with `self.update_metadata(draft, metadata, reviewer)`

corbinSanders replied:
🆗

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/nodes/serializers.py:
```diff
@@ -1466,7 +1458,7 @@ def to_internal_value(self, registration_schema_id):
 
 
-class DraftRegistrationSerializer(JSONAPISerializer):
+class DraftRegistrationSerializerLegacy(JSONAPISerializer):
```
Change to `DraftRegistrationLegacySerializer`

corbinSanders replied:
🆗

corbinSanders commented on api/nodes/views.py:
```diff
@@ -674,7 +692,7 @@ def perform_create(self, serializer):
         """
         # On creation, make sure that current user is the creator
-        draft_id = self.request.data.get('draft_registration', None) or self.request.data.get('draft_registration_id', None)
+        draft_id = self.request.data.get('draft_registration', None)
```
Place or statement back in

corbinSanders replied:
✅

corbinSanders commented on api/nodes/views.py:
```diff
@@ -2051,7 +2075,6 @@ def get_object(self):
     def perform_destroy(self, link):
         assert isinstance(link, PrivateLink), 'link must be a PrivateLink'
         link.is_deleted = True
-        link.deleted = timezone.now()
```
Place is_deleted back in.

corbinSanders replied:
✅

UdayVarkhedkar commented on api/registrations/serializers.py:
```diff
@@ -668,6 +666,21 @@ def _is_attached_file_valid(self, file_metadata, registering):
         return True
 
+
+class RegistrationCreateSerializerOldWorkflow(RegistrationCreateSerializer):
```
rename `RegistrationCreateLegacySerializer`

corbinSanders replied:
✅

UdayVarkhedkar commented on api/registrations/serializers.py:
```diff
@@ -13,9 +13,9 @@
-from api.nodes.serializers import NodeSerializer, NodeStorageProviderSerializer
-from api.nodes.serializers import NodeLinksSerializer, NodeLicenseSerializer, update_institutions
-from api.nodes.serializers import NodeContributorsSerializer, NodeLicenseRelationshipField, RegistrationProviderRelationshipField, get_license_details
+from api.nodes.serializers import NodeSerializer, NodeStorageProviderSerializer, NodeLicenseRelationshipField
```
Combine with other api.nodes.serializers imports

corbinSanders replied:
👌

UdayVarkhedkar commented on api/registrations/serializers.py:
```diff
@@ -618,7 +618,7 @@ def _find_orphan_files(self, registering, draft):
         files = find_selected_files(draft.registration_schema, draft.registration_metadata)
         orphan_files = []
-        for key, value in files.items():
+        for key, value in list(files.items()):
```
Check if `list()` is necessary

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/registrations/serializers.py:
```diff
@@ -631,11 +631,9 @@ def _is_attached_file_valid(self, file_metadata, registering):
         Verifying that nodeId in the file_metadata is one of the files we're registering. Verify
         that selectedFileName is the name of a file on the node.  Verify that the sha256 matches
         a version on that file.
-
```
Add back - removed in schemablocks merge

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/registrations/serializers.py:
```diff
@@ -631,11 +631,9 @@ def _is_attached_file_valid(self, file_metadata, registering):
         Validation of file information on registration_metadata.  Theoretically, the file information
         on registration_responses does not have to be valid, so we enforce their accuracy here,
         to ensure file links load properly.
-
```
Add back - removed in schemablocks merge

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/registrations/serializers.py:
```diff
@@ -13,9 +13,9 @@
+from api.nodes.serializers import NodeSerializer, NodeStorageProviderSerializer, NodeLicenseRelationshipField
+from api.nodes.serializers import NodeLinksSerializer, update_institutions, NodeLicenseSerializer
+from api.nodes.serializers import NodeContributorsSerializer, RegistrationProviderRelationshipField, get_license_details
 from api.base.serializers import (
```
unroll into separate lines

corbinSanders replied:
🆗

UdayVarkhedkar commented on api/registrations/views.py:
```diff
@@ -86,7 +89,7 @@ def get_node(self, check_object_permissions=True):
 
 
-class RegistrationList(JSONAPIBaseView, generics.ListAPIView, bulk_views.BulkUpdateJSONAPIView, NodesFilterMixin):
+class RegistrationList(JSONAPIBaseView, generics.ListCreateAPIView, bulk_views.BulkUpdateJSONAPIView, NodesFilterMixin, DraftMixin):
```
Possibly replace `DraftMixin` with draft-registrations-specific permissions classes

UdayVarkhedkar commented on api/nodes/views.py:
```diff
@@ -175,16 +178,24 @@ def get_node(self, check_object_permissions=True):
 
 class DraftMixin(object):
```
Look into renaming to something more representative given multiple draft models

UdayVarkhedkar commented on api/registrations/views.py:
```diff
@@ -160,6 +165,28 @@ def get_queryset(self):
+
+        # A user must be an admin contributor on the node (not group member), and have
+        # admin perms on the draft to register
+        if node.is_admin_contributor(user) and draft.has_permission(user, ADMIN):
```
Roll up into permissions class

UdayVarkhedkar commented on api/users/views.py:
```diff
@@ -83,7 +84,6 @@
     OSFUser,
     Email,
 )
-from osf.utils import permissions
```
Check if necessary import

corbinSanders replied:
Not seeing any utilization of this.

corbinSanders commented on api/users/views.py:
```diff
@@ -576,7 +576,7 @@ class UserIdentitiesList(JSONAPIBaseView, generics.ListAPIView, UserMixin):
         user = self.get_user()
         identities = []
-        for key, value in user.external_identity.items():
+        for key, value in list(user.external_identity.items()):
```
Check if `list()` is necessary here?

corbinSanders replied:
👌

UdayVarkhedkar commented on api/users/views.py:
```diff
@@ -496,21 +496,21 @@ class UserDraftRegistrations(JSONAPIBaseView, generics.ListAPIView, UserMixin):
-        )
+        requested_user = self.get_user()
+        requesting_user = self.request.user
+        if requesting_user.is_anonymous:
```
Should perhaps be replaced with a KnownUser permissions class

UdayVarkhedkar commented on api/users/views.py:
```diff
@@ -496,21 +496,21 @@ class UserDraftRegistrations(JSONAPIBaseView, generics.ListAPIView, UserMixin):
-            branched_from__in=list(node_qs),
-            deleted__isnull=True,
-        )
+        requested_user = self.get_user()
```
Remove `requested_user` - Does not offer meaningful functionality

corbinSanders replied:
🆗

UdayVarkhedkar commented on osf/models/mixins.py:
```diff
@@ -2169,6 +2169,7 @@ def copy_editable_fields(self, resource, auth=None, alternative_resource=None, s
 
         if save:
             self.save()
+        return
```
remove empty return

corbinSanders replied:
👌

UdayVarkhedkar commented on osf_tests/test_draft_registration.py:
```diff
@@ -385,9 +385,9 @@ def model(self):
+class TestDraftRegistrationContributorMethods():
     @pytest.fixture()
-    def draft_registration(self, project):
+    def resource(self, project):
```
Look into use of resource fixture vs draft_registration fixture

corbinSanders replied:
Ended up removing several of these fixtures as they weren't being used.

UdayVarkhedkar commented on website/project/views/contributor.py:
```diff
@@ -555,6 +555,8 @@ def notify_added_contributor(node, contributor, auth=None, throttle=None, email_
             else:
                 logo = preprint_provider._id
+        elif email_template == 'draft_registration':
+            email_template = getattr(mails, 'CONTRIBUTOR_ADDED_DRAFT_REGISTRATION'.format(email_template.upper()))
```
remove `.format(email_template.upper())`

corbinSanders replied:
🆗

UdayVarkhedkar commented on website/project/views/contributor.py:
```diff
@@ -541,8 +541,8 @@ def notify_added_contributor(node, contributor, auth=None, throttle=None, email_
     # Email users for projects, or for components where they are not contributors on the parent node.
-    if contributor.is_registered and (isinstance(node, Preprint) or
-            (not node.parent_node or (node.parent_node and not node.parent_node.is_contributor(contributor)))):
+    if contributor.is_registered and (((isinstance(node, (Preprint, DraftRegistration))) or
```
minimize parenthesis usage

corbinSanders replied:
✅

UdayVarkhedkar commented on website/settings/local-dist.py:
```diff
@@ -51,6 +51,10 @@
         'server': 'http://{}:4203/'.format(EMBER_DOMAIN),
         'path': '/reviews/'
     },
+    'registries': {
```
Remove - should be unnecessary given registries has been pulled into ember_osf_web

corbinSanders replied:
👌

UdayVarkhedkar commented on website/views.py:
```diff
@@ -36,6 +36,7 @@
 
 logger = logging.getLogger(__name__)
 preprints_dir = os.path.abspath(os.path.join(os.getcwd(), EXTERNAL_EMBER_APPS['preprints']['path']))
+registries_dir = os.path.abspath(os.path.join(os.getcwd(), EXTERNAL_EMBER_APPS['registries']['path']))
```
Remove - unnecessary because registries is wrapped into ember_osf_web

corbinSanders replied:
👌

UdayVarkhedkar commented on website/templates/emails/contributor_added_draft_registration.html.mako:
```diff
@@ -0,0 +1,26 @@
+    %>
+    Hello ${user.fullname},<br>
+    <br>
+    ${referrer_name + ' has added you' if referrer_name else 'You have been added'} as a contributor to the draft registration "${node.title}" on the Open Science Framework: ${node.absolute_url}<br>
```
Change to 'on OSF'

corbinSanders replied:
👌

UdayVarkhedkar commented on website/templates/emails/contributor_added_draft_registration.html.mako:
```diff
@@ -0,0 +1,26 @@
+    <br>
+    Sincerely,<br>
+    <br>
+    Open Science Framework Robot<br>
```
Change to 'OSF Robot'

corbinSanders replied:
👌

aaxelb commented on api_tests/draft_registrations/views/test_draft_registration_contributor_detail.py:
```diff
@@ -0,0 +1,268 @@
+            API_BASE, project._id, contrib._id)
+
+    def test_change_contributor_non_admin_osf_group_member_auth(self, project, contrib):
+        # Overrides TestNodeContributorUpdate - drafts have no perms
```
```suggestion
        # Overrides TestNodeContributorUpdate - drafts have no group perms
```

corbinSanders replied:
🆗

aaxelb commented on api_tests/draft_registrations/views/test_draft_registration_contributor_detail.py:
```diff
@@ -0,0 +1,268 @@
+        return
+
+    def test_change_contributor_admin_osf_group_permissions(self, project, contrib):
+        # Overrides TestNodeContributorUpdate - drafts have no perms
```
```suggestion
        # Overrides TestNodeContributorUpdate - drafts have no group perms
```

corbinSanders replied:
🆗

cslzchen committed:
New institution: Concordia College

aaxelb commented on api_tests/nodes/views/test_node_contributors_list.py:
```diff
@@ -630,62 +629,59 @@ def test_add_contributor_dont_expose_email(
     def test_add_contributor_is_visible_by_default(
             self, app, user, user_two, project_public,
             data_user_two, url_public):
-        with assert_latest_log(NodeLog.CONTRIB_ADDED, project_public):
```
i assume these were removed throughout this file because `DraftRegistration`s don't have `NodeLog`s, but are there other tests to assert that logs are correctly added to nodes? doesn't seem like a good thing to simply remove.

corbinSanders replied:
There are still log tests in `test_node_contributors_detail.py`.

Commit:
Making suggested fixes

aaxelb commented on api_tests/draft_registrations/views/test_draft_registration_contributor_detail.py:
```diff
@@ -0,0 +1,268 @@
+class TestDraftContributorDetail(TestContributorDetail):
+    @pytest.fixture()
+    def project_public(self, user, title, description, category):
+        # Defining "project public" as a draft reg, overriding TestContributorDetail
```
ok... this does seem like the quickest route to reusing the existing contributor tests, but it also sows seeds of confusion and it'd be cleaner to refactor the existing tests to be more reusable...

that's probably more a consideration for the future than something to take on right now, though

UdayVarkhedkar replied:
The entanglement of contributors and nodes makes cleaning up the tests to be more reusable a bit more complicated. I agree with you but wanted to bring it up as a byproduct of the node-centric nature of some of our models.

aaxelb replied:
makes sense -- maybe we should consider adding a `contributor-basket` to go along with `file-basket` (since both are much more complex than a basic to-many relation)

aaxelb commented on api_tests/nodes/views/test_node_list.py:
```diff
@@ -3392,7 +3403,7 @@ def test_skip_uneditable_bulk_update(
         skipped = res.json['errors']
         assert_equals(
-            [edited[0]['id'], edited[1]['id']],
+            [edited[1]['id'], edited[0]['id']],
```
is there a reason the order has flipped (here and below)?

if we don't actually care about order, we should use a set (or otherwise assert without order)

corbinSanders replied:
This was fixed. It was switched because I was importing the wrong assert_equals. `assert_items_equal` was renamed to `assert_equals` during the py3 upgrade. I did not know this until recently.

aaxelb commented on api_tests/nodes/views/test_node_sparse_fieldsets.py:
```diff
@@ -43,15 +43,15 @@ def test_node_sparse_fields_list(
 
         assert node_json['attributes'] == {}
-        assert set(node_json.keys()) == set(
+        assert set(list(node_json.keys())) == set(
```
`set(list(...))` does nothing helpful compared to `set(...)` (throughout this file) -- it seems like we're a little too afraid of py3's view objects for some reason

corbinSanders replied:
I removed this `list(...)` usage.

aaxelb reviewed (commented):
tests!

Commit:
First round of tests

Commit:
Fixing api tests broken by rebase

cslzchen committed:
Update Boys Town to use SAML auth on the test server

Commit:
Fixing draft registration id related things

mfraezz committed:
Merge branch 'hotfix/20.3.1'

mfraezz committed:
Merge branch 'hotfix/20.3.1' into develop

mfraezz committed:
Merge branch 'hotfix/20.3.2'

mfraezz committed:
Merge branch 'hotfix/20.3.2' into develop

Commit:
Fixing URLs in tests

UdayVarkhedkar committed:
Updates permissions for draft_registration api endpoints

Commit:
Making recommended fixes to tests

UdayVarkhedkar committed:
Replaces uses of Faker('word') with factory.Sequence

UdayVarkhedkar committed:
Updates scope factory and scope urls to improve scope testing

UdayVarkhedkar committed:
Merge pull request #5 from corbinSanders/feature/draft_registration_api_endpoints_combined

Feature/draft registration api endpoints combined

Commit:
Fixing namespace issues

UdayVarkhedkar committed:
Merge pull request #6 from corbinSanders/feature/draft_registration_api_endpoints_combined

Fixing namespace issues

Johnetordoff committed:
resolve `six` conflict that was breaking Travis

Johnetordoff committed:
drive-by: address depreciation warnings for travis

Commit:
Fixing branched from tests

mfraezz committed:
Merge branch 'hotfix/20.3.3'

mfraezz committed:
Merge branch 'hotfix/20.3.3' into develop

UdayVarkhedkar committed:
[ENG-1206] Backend Registration Workflow - Breaks out Models and Migrations (#9229)

## Purpose

Break out the models and migrations code from Dawn's [DraftNode Backend Registration Workflow PR](https://github.com/CenterForOpenScience/osf.io/pull/9093) into a separate branch to serve as foundation for updating and completing the backend changes necessary for the new registrations workflow. 

## Changes
* Rebases Dawn's code changes on top of `develop` and fix merge conflicts with schemablocks code

### Model Changes
* Creates `DraftNode` model
* Alters `DraftRegistration` model to add editable fields
* Creates `DraftRegistrationContributor` model to handle permissions for DraftRegistrations
* Adds `DraftRegistrationUserObjectPermission`, `DraftRegistrationGroupObjectPermission` to help handle permissions for DraftRegistrations
* Creates new `EditableFieldMixin` model
  * Creates mixin models such as `TitleMixin`, `DescriptionMixin`, etc. for editable fields
  * Edits mixin model parents for classes (i.e. Preprints) accordingly
* Updates `DraftRegistrationLog` model to allow for more detailed logging messages around editable fields

### Migrations
* 0195 - Migration for model changes enumerated above
* 0196 - Updates DraftRegistration permissions

### Model Testing
* Breaks out `DraftRegistration` tests
* Adds testing for `DraftNode`
* Adds test for Node registration with edited draft registration fields
* Replaces tests for `TestNodeUpdate` with tests for `EditableFieldMixin`
* Moves tests for `NodeLicenses` to `osf_tests/test_node`
* Adds test for merging `DraftRegistrations`

## QA Notes

QA Testing will be done for DraftNode and the new backend registration workflow overall once complete. Testing for this PR will be dependent on good unit testing. 

## Documentation

N/A

## Side Effects

All current DraftRegistration and Registration creation processes should be unaffected by these changes. Any possible side effects of these changes on the existing workflow should be avoided.

## Ticket

https://openscience.atlassian.net/browse/ENG-1206

UdayVarkhedkar committed:
Merge branch 'develop' into feature/draft_registration_api_endpoints_combined

UdayVarkhedkar committed:
Merge pull request #7 from corbinSanders/feature/draft_registration_api_endpoints_combined

Fixing branched_from tests

Commit:
Fixing self links in draft registration endpoint

UdayVarkhedkar committed:
Fixes permissions tests for draft_registrations

UdayVarkhedkar committed:
Removes outdated pre-reg challenge tests from draft_registrations

UdayVarkhedkar committed:
Merge pull request #8 from corbinSanders/feature/draft_registration_api_endpoints_combined

Fixing self links in draft registration endpoint

fabmiz committed:
Return proper type for user::user-settings relationship (#9287)

## Purpose

With recent  ember-data upgrade, this inconsistency in resource types is not taken lightly and causes undesirable issues on the front-end.

## Changes

- Update base serializer to return `user-settings` (instead of `users`) as type for user_settings relationship.

## QA Notes
- This change should fix issues outlined in on the frontend.


## Ticket
[ENG-1462](https://openscience.atlassian.net/browse/ENG-1462)
[ENG-1482](https://openscience.atlassian.net/browse/ENG-1482)

brianjgeiger committed:
Fix the b fields (#9290)

## Purpose

There was a bit of migration missing from develop. This adds it in.

## Changes

Modify the models that had `b''` to normal strings.

## QA Notes

  - Does this change require a data migration? If so, what data will we migrate? _This will alter some field structure, but not data as such._
  - What is the level of risk? _Low_
    - Any permissions code touched? _Nope_
    - Is this an additive or subtractive change, other? _A plain change_
  - How can QA verify? (Through UI, API, AdminApp or AdminAdminApp?) _Make sure that the category, description, and title fields still work for draft registrations._
    - If verifying through API, what's the new version? Please include the endpoints in PR notes or Dev docs. _N/A_
  - What features or workflows might this change impact? _Draft registration_
  - How will this impact performance? _It will not_


## Side Effects

Shouldn't be

## Ticket

N/A

Commit:
Fixing set unions in can view

Commit:
Adding in tests for failing case

mfraezz committed:
Merge branch 'hotfix/20.3.4' into develop

UdayVarkhedkar committed:
Merge branch 'develop' into feature/draft_registration_api_endpoints_combined

UdayVarkhedkar committed:
Merge branch 'feature/draft_registration_api_endpoints_combined' of https://github.com/UdayVarkhedkar/osf.io into feature/draft_registration_api_endpoints_combined

UdayVarkhedkar committed:
Smaller syntax improvements

UdayVarkhedkar committed:
Updates use of __name__ to modify __class__ rather than a class instance

UdayVarkhedkar committed:
Fixes DraftRegistration.branched_from_type
